fvm_actor_utils/src/receiver/frc46.rs

```diff
@@ -0,0 +1,27 @@
+use frc42_dispatch::method_hash;
+use fvm_ipld_encoding::tuple::{Deserialize_tuple, Serialize_tuple};
+use fvm_ipld_encoding::{Cbor, RawBytes};
+use fvm_shared::{econ::TokenAmount, ActorID};
+
+use super::ReceiverType;
+
+pub const FRC46_TOKEN_TYPE: ReceiverType = method_hash!("FRC46") as u32;
+
+/// Receive parameters for an FRC46 token
+#[derive(Serialize_tuple, Deserialize_tuple, PartialEq, Eq, Clone, Debug)]
+pub struct FRC46TokenReceived {
```

Comment:
I kinda think this should go in the FRC46 package rather than here, but it's a weak desire, and I haven't traced through to understand the resulting deps. Certainly we shouldn't end up with this crate depending on frc46.

Reply:
I think it only needs to be available for the `new_frc46` helper (see https://github.com/helix-onchain/filecoin/pull/109/files#diff-8dc699e4ff6e5f5e0e14a7ddec8ef0414438e44cd18e6eccc1ecaa3ff4f2f6d9R106) but that could be resolved by implementing the function in a separate helper trait inside the `frc46_token` crate. Then token library users would simply `use frc46_token::FRC46Receiver` (or whatever we name it) to have that method available. Might be worth revisiting the idea to keep the token-specific parts separate as the receiver itself is meant to be universal

Reply:
I've split this out now and will follow the same pattern for NFTs rather than bloating the universal package